core/modules/qdisp/Executive.cc

```diff
@@ -352,33 +369,36 @@ int Executive::getNumInflight() {
 }
 
 std::string Executive::getProgressDesc() const {
-    std::lock_guard<std::mutex> lock(_statusesMutex);
     std::ostringstream os;
-    std::for_each(_statuses.begin(), _statuses.end(), printMapEntry(os, "\n"));
-    LOGF_ERROR("%1%" % os.str());
-    return os.str();
+    {
+        std::lock_guard<std::mutex> lock(_statusesMutex);
+        std::for_each(_statuses.begin(), _statuses.end(), printMapEntry(os, "\n"));
+    }
+    auto const & msg_progress = os.str();
+    LOGF(getLogger(), LOG_LVL_ERROR, "%1%" % msg_progress);
+    return std::move(msg_progress);
```

Comment:
My experience with std::move is very limited and I'm not sure that it does anything useful here. I'd probably just rely on good old NRVO:

    std::string msg_progress = os.str();
    ....
    return msg_progress;